src/helpers/CBV/tests/ObjectList.test.tsx

```diff
@@ -0,0 +1,130 @@
+
+/** ObjectList options */
+const objectListOptions = {
+  actionCreator: sendMessage,
+  dispatchPropName: 'actionCreator',
+  returnPropName: 'messages',
+  selector: selectAllMessages,
+};
+
+const ClassBasedView = new ObjectList<
+  Message,
+  SendMessageAction,
+  typeof selectAllMessages,
+  TestProps
+>(TestComponent, objectListOptions);
+
+describe('cbv/ObjectList', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    store.dispatch(removeMessagesAction());
+  });
+  it('getMapStateToProps works as expected', () => {
+    const mapStateToProps = ClassBasedView.getMapStateToProps();
+
+    expect(typeof mapStateToProps === 'function').toBeTruthy();
+
+    expect(mapStateToProps(store.getState(), {} as any)).toEqual({ messages: [] });
+
+    store.dispatch(sendMessage({ user: 'bob', message: 'hello' }));
+    store.dispatch(sendMessage({ user: 'bobbie', message: 'hello hello' }));
+
+    expect(mapStateToProps(store.getState(), {} as any)).toEqual({
+      messages: [{ user: 'bob', message: 'hello' }, { user: 'bobbie', message: 'hello hello' }],
+    });
+  });
+
+  it('getMapDispatchToProps works as expected', () => {
+    const mapDispatchToProps = ClassBasedView.getMapDispatchToProps();
+    expect(typeof mapDispatchToProps === 'function').toBeFalsy();
+    expect(mapDispatchToProps).toEqual({ actionCreator: sendMessage });
+  });
+
+  it('getHOC works as expected', () => {
+    const HoC = ClassBasedView.getHOC();
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <HoC />
+      </Provider>
+    );
+
+    const expected = {
+      actionCreator: sendMessage,
+      objectList: [],
+    };
+
+    expect(wrapper.find('HoC').props()).toEqual(expected);
+    expect(wrapper.find('HoC>TestComponent').props()).toEqual(expected);
+  });
+
+  it('render and getConnectedHOC works as expected', () => {
+    const ConnectedHOCComponent = ClassBasedView.getConnectedHOC();
+    const ConnectedTestComponent = ClassBasedView.render();
+
+    const toTest = [ConnectedHOCComponent, ConnectedTestComponent];
+
+    for (const Element of toTest) {
+      store.dispatch(removeMessagesAction());
+      store.dispatch(sendMessage({ user: 'bob', message: 'hello' }));
+      store.dispatch(sendMessage({ user: 'bobbie', message: 'hello hello' }));
+
+      const wrapper = mount(
+        <Provider store={store}>
+          <Element />
+        </Provider>
+      );
+
+      const expected = {
+        actionCreator: sendMessage,
+        messages: [{ user: 'bob', message: 'hello' }, { user: 'bobbie', message: 'hello hello' }],
+        objectList: [],
+      };
+
+      expect(wrapper.find('Connect(HoC)').props()).toEqual({});
+      expect(wrapper.find('Connect(HoC)>HoC').props()).toEqual({
+        ...expected,
+        actionCreator: expect.any(Function),
+      });
+      expect(wrapper.find('Connect(HoC)>HoC>TestComponent').props()).toEqual({
+        ...expected,
+        actionCreator: expect.any(Function),
+      });
```

Comment:
👏 Nice props testing! If the `TestComponent` renders a list of objects, could we add a line or two testing that `TestComponent` is rendering correctly with the props it's given? If the component renders a simple list, maybe something like:

```
expect(wrapper.find('li').length).toEqual(2)
```

Not sure if this actually increases test coverage though?

Reply:
It will not improve the test coverage (its already 100%) but I think it is useful for clarity.  I'll add this.